Added Oracle compatibility of the NOT NULL syntax.

plus33 opened the pull request:
Oracles NOT NULL syntax can have optional ENABLE VALIDATE added e.g  
  create table T (C int NOT NULL ENABLE VALIDATE)
Some SQL-generators produce such SQL and H2 can now parse it with this
change.

plus33 committed:
Added Oracle compatibility of the NOT NULL syntax. 

Oracles NOT NULL syntax can have optional ENABLE VALIDATE added e.g  
  create table T (C int NOT NULL ENABLE VALIDATE)
Some SQL-generators produce such SQL and H2 can now parse it with this
change.

grandinj commented on h2/src/main/org/h2/command/Parser.java:
```diff
@@ -6352,11 +6372,14 @@ private CreateTable parseCreateTable(boolean temp, boolean globalTemp,
-                            read("NULL");
+
+                        NULL_CONSTRAINT nullConstraint = parseNotNullConstraint();
+                        switch (nullConstraint) {
```
can just use an if statement here

grandinj commented on h2/src/main/org/h2/command/Parser.java:
```diff
@@ -6459,6 +6482,34 @@ private CreateTable parseCreateTable(boolean temp, boolean globalTemp,
         return command;
     }
 
+    private enum NULL_CONSTRAINT {
```
please call this NullConstraintType

grandinj commented on h2/src/main/org/h2/command/Parser.java:
```diff
@@ -6459,6 +6482,34 @@ private CreateTable parseCreateTable(boolean temp, boolean globalTemp,
+                read("NULL");
+                nullConstraint = NULL_CONSTRAINT.NULL_IS_ALLOWED;
+            }
+            if (readIf("ENABLE")) {
```
please check that we are in Oracle mode here

grandinj reviewed (changes requested):
looks generally good. I think you're going to need to merge with master, some conflicts are present

plus33 committed:
Changes based on feedback form Noel Grandin

* Renamed enum NULL_CONSTRAINT to NullConstraintType
* Replaced some switches with if-statement
* Check for Oracle-Mode when parsing Oracles-SQLs NOT NULL ENABLE
VALIDATE syntax. NOTE: 
* Added DbTypeEnum-class to Mode to have a String-free dbType check
featuring the method Database.isDbTypeOneOf(DbTypeEnum... dbTypes)

plus33 committed:
* Moved Oracle specific SQL tests to TestCompatibilityOracle

plus33 committed:
Merge remote-tracking branch 'origin/master'

grandinj commented on h2/src/main/org/h2/engine/Mode.java:
```diff
@@ -23,7 +23,25 @@
     static final String REGULAR = "REGULAR";
 
-    private static final HashMap<String, Mode> MODES = New.hashMap();
+    public static enum DbTypeEnum {
```
you're going to need to merge to master, someone else already added a mode enum :-)
